gpkit/tools/tools.py

```diff
@@ -113,3 +115,63 @@ def mdmake(filename, make_tex=True):
         with open(filename+".tex.md", "w") as f:
             f.write(texmd)
     return open(filename+".py")
+
+
+def bound_all_variables(varkeys, lb, ub):
+    "Returns model with additional constraints bounding all free variables"
+    constraints = []
+    for varkey in varkeys:
+        units = varkey.descr.get("units", 1)
+        constraints.append([ub*units >= Variable(**varkey.descr),
+                            Variable(**varkey.descr) >= lb*units])
+    return constraints
+
+
+class BoundedConstraintSet(ConstraintSet):
+    "Bounds contained variables so as to ensure dual feasibility."
+
+    def __init__(self, constraints, substitutions=None, verbosity=1,
+                 eps=1e-30, lower=None, upper=None):
+        self.verbosity = verbosity
+        self.lb = lower if lower else eps
+        self.ub = upper if upper else 1/eps
+        constraints = ConstraintSet([constraints])
+        self.bounded_varkeys = tuple(vk for vk in constraints.varkeys
+                                     if "value" not in vk.descr)
+        bounding_constraints = bound_all_variables(self.bounded_varkeys,
+                                                   self.lb, self.ub)
+        constraints = [bounding_constraints, constraints]
+        super(BoundedConstraintSet, self).__init__(constraints, substitutions)
+
+    def sens_from_dual(self, las, nus):
+        "Return sensitivities while capturing the relevant lambdas"
+        self.bound_las = las[:2*len(self.bounded_varkeys)]
+        return super(BoundedConstraintSet, self).sens_from_dual(las, nus)
+
+    def process_result(self, result):
+        "Creates (and potentially prints) a dictionary of unbounded variables."
```

Comment:
description would be more precise if it said something like `"... a dictionary of variables affected by the bounds"` ?